stacker/blueprints/base.py

```diff
@@ -152,6 +160,60 @@ def setup_parameters(self):
             p = build_parameter(param, attrs)
             t.add_parameter(p)
 
+    def defined_variables(self):
+        """Return a dictionary of variables defined by the blueprint.
+
+        By default, this will just return the values from `VARIABLES`, but this
+        makes it easy for subclasses to add variables.
+
+        Returns:
+            dict: variables defined by the blueprint
+
+        """
+        return getattr(self, "VARIABLES", {})
+
+    def get_variables(self):
+        """Return a dictionary of variables available to the template.
+
+        These variables will have been defined within `VARIABLES` or
+        `self.defined_variables`. Any variable value that contains a lookup
+        will have been resolved.
+
+        Returns:
+            dict: variables available to the template
+
+        """
+        if self.resolved_variables is None:
+            raise UnresolvedVariables(self)
+        return self.resolved_variables
+
+    def resolve_variables(self, variables):
+        """Resolve the values of the blueprint variables.
+
+        This will resolve the values of the `VARIABLES` with values from the
+        env file, the config, and any lookups resolved.
+
+        Args:
+            variables (list): list of variables
```

Comment:
This is a list of a specific class, right?  Can you use the `:class:` syntax to point at the actual class that you want a list of?  Right now it's hard to tell what a `list of variables` means.